test/test_client_db_hooks.js

```diff
@@ -0,0 +1,62 @@
+import db         from '../src/models/db';
+import * as CONST from './constants';
+
+db().then(models => {
+  describe('db hooks', () => {
+    beforeEach(() => {
+      return models.sequelize.transaction(transaction => {
+        return Promise.all(Object.keys(CONST.entities).map(name => {
+          return models[name].destroy({ transaction, where: {} });
+        }));
+      });
+    });
+
+    Object.keys(CONST.entities).forEach(entity => {
+      it(entity + ' db "create" hooks', done => {
+        models[entity].create(CONST[entity + 'Entity']).then(instance => {
+          [
+            'beforeValidate',
+            'afterValidate',
+            'beforeCreate',
+            'afterCreate'
+          ].forEach(hook => {
+            hook.should.be.equal(instance[hook]);
+          });
+        });
+        done();
+      })
+    });
+
+    // This test does not cause the before/afterDestroy hooks to be run.
+    // From Sequelize documentation:
+    // (http://docs.sequelizejs.com/en/latest/docs/hooks/)
+    // "The only way to call beforeDestroy/afterDestroy hooks are on
+    // associations with `onDelete: 'cascade'` and the option `hooks: true`"
+    // TBD It seems our ST API implementation does not exercise these hooks.
+    // Perhaps we should not implement them.
+    xit('db "destroy" hooks', done => {
+      models[CONST.things].create(CONST.ThingsEntity).then(instance => {
+        const id = instance.id;
+        models[CONST.things].destroy({ where: { id } }).then(() => {
+          done();
+        });
+      });
+    })
+
+    xit('db "update" hooks', done => {
+      // TODO The 'before/afterUpdate' hooks are not being run during this
+      // test. Why not?
```

Comment:
I saw that the 'before/afterUpdate' hooks were running during other tests but during this test they do not run. I haven't been able to figure out why they don't run during this test.